application/src/main/java/org/thingsboard/server/service/install/update/EntityPaginatedUpdater.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright © 2016-2021 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.service.install.update;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.thingsboard.server.common.data.SearchTextBased;
+import org.thingsboard.server.common.data.id.UUIDBased;
+import org.thingsboard.server.common.data.page.PageLink;
+import org.thingsboard.server.dao.DaoUtil;
+import org.thingsboard.server.dao.model.BaseSqlEntity;
+
+public abstract class EntityPaginatedUpdater<I, D extends BaseSqlEntity<? extends SearchTextBased<? extends UUIDBased>>> {
```

Comment:
Why not just change from
```java
public abstract class PaginatedUpdater<I, D extends SearchTextBased<? extends UUIDBased>> {
```
to 
```java 
public abstract class PaginatedUpdater<I, D> {
```

and execute DaoUtil.toPageable(pageLink) inside the findEntities..